src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +230,212 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
       }
-      String token = response.statusCode() == 200
-        ? response.getHeader(XOkapiHeaders.TOKEN)
-        : response.bodyAsJsonObject().getString("token");
-      if (token == null) {
-        throw new RuntimeException(String.format("Got response %s fetching token, but content is null",
-          response.statusCode()));
-      }
-      logger.debug("Got token " + token + " from authz");
-      return token;
+
+      return response.bodyAsJsonObject();
     });
   }
 
-  private Future<String> fetchRefreshToken(String userId, String sub, String tenant,
-    String okapiURL, String requestToken) {
-    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + "/refreshtoken");
+  private Future<HttpResponse<Buffer>> logout(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ENDPOINT);
+
     request.putHeader(XOkapiHeaders.TENANT, tenant)
-      .putHeader(XOkapiHeaders.TOKEN, requestToken);
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
 
-    JsonObject payload = new JsonObject().put("userId", userId).put("sub", sub);
-    return request
-      .expect(ResponsePredicate.SC_CREATED)
-      .expect(ResponsePredicate.JSON)
-      .sendJsonObject(payload)
-      .map(response -> response.bodyAsJsonObject().getString("refreshToken")
-      );
+      return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private Future<HttpResponse<Buffer>> logoutAll(String tenant,
+      String okapiURL, String accessToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ALL_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.send();
+  }
+
+  private Future<HttpResponse<Buffer>> fetchRefreshToken(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_REFRESH_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private boolean usesTokenSignLegacy(String endpoint) {
+    return endpoint.equals(TOKEN_SIGN_ENDPOINT_LEGACY);
+  }
+
+
+  // For documentation of the cookie attributes please see:
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+  private String refreshTokenCookie(String refreshToken, String refreshTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(refreshTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
```

Comment:
RFC 6265 mandates that MaxAge is >= 1: https://datatracker.ietf.org/doc/html/rfc6265#page-9

Consider Expires with DateTimeFormatter.RFC_1123_DATE_TIME.

Reply:
Not sure I'm understanding this one. With this code `MaxAge` will be > than 1. Also the Vertx Cookie adds an `Expires` attribute automatically based on what is provided with  `MaxAge`.

Reply:
How can you ensure that ttlSeconds is > 0? refreshTokenExpiration might be in the past.